drivers/modem/modem_iface_uart.c

```diff
@@ -228,3 +229,27 @@ int modem_iface_uart_init(struct modem_iface *iface,
 
 	return 0;
 }
+
+void modem_iface_sleep(struct modem_iface *iface)
+{
+	uart_irq_rx_disable(iface->dev);
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+	device_set_power_state(iface->dev,
+			       DEVICE_PM_LOW_POWER_STATE,
```

Comment:
Why do we use `DEVICE_PM_LOW_POWER_STATE` here? Shouldn't we have a possibility to pass any power state passed as function argument? E.g. when driver will want to handle DEVICE_PM_OFF_STATE, then it most probably wants to use the same for UART.